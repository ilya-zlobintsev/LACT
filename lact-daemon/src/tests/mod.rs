use insta::assert_json_snapshot;

use crate::{config::Config, server::handler::Handler};
use std::{fs, path::PathBuf};

#[tokio::test]
async fn snapshot_everything() {
    let test_data_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("src/tests/data");

    for vendor_dir in fs::read_dir(test_data_dir).unwrap().flatten() {
        for device_dir in fs::read_dir(vendor_dir.path()).unwrap().flatten() {
            let test_key = format!(
                "{}/{}",
                vendor_dir.file_name().to_string_lossy(),
                device_dir.file_name().to_string_lossy()
            );

            let handler = Handler::with_base_path(&device_dir.path(), Config::default(), true)
                .await
                .unwrap();
            let mut device_info = handler
                .generate_snapshot_device_info()
                .into_values()
                .next()
                .unwrap();
            // Remove vulkan information, as it can only be fetched from the current running system and affects snapshot output
            device_info
                .get_mut("info")
                .unwrap()
                .as_object_mut()
                .unwrap()
                .remove("vulkan_info");

            assert_json_snapshot!(test_key, device_info);
        }
    }
}
