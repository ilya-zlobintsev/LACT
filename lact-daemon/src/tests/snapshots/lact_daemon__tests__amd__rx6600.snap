---
source: lact-daemon/src/tests/mod.rs
expression: device_info
---
{
  "clocks_info": {
    "max_mclk": 925,
    "max_sclk": 2700,
    "table": {
      "type": "amd",
      "value": {
        "data": {
          "current_mclk_range": {
            "max": 925,
            "min": 97
          },
          "current_sclk_range": {
            "max": 2700,
            "min": 2600
          },
          "od_range": {
            "curve_sclk_points": [],
            "curve_voltage_points": [],
            "mclk": {
              "max": 950,
              "min": 674
            },
            "sclk": {
              "max": 2900,
              "min": 500
            },
            "sclk_offset": null,
            "voltage_offset": null
          },
          "rdna4_sclk_offset_workaround": false,
          "sclk_offset": null,
          "vddc_curve": [],
          "voltage_offset": -100
        },
        "kind": "vega20"
      }
    }
  },
  "info": {
    "driver": "amdgpu",
    "link_info": {
      "current_speed": "16.0 GT/s PCIe",
      "current_width": "16"
    },
    "pci_info": {
      "device_pci_info": {
        "model": "Navi 23 [Radeon RX 6600/6600 XT/6600M]",
        "model_id": "73FF",
        "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
        "vendor_id": "1002"
      },
      "subsystem_pci_info": {
        "model": "GIGABYTE RX 6600 Eagle",
        "model_id": "2334",
        "vendor": "Gigabyte Technology Co., Ltd",
        "vendor_id": "1458"
      }
    },
    "vbios_version": "113-D534-R66E",
    "vulkan_info": {
      "api_version": "",
      "device_name": "",
      "driver": {
        "version": 0
      },
      "enabled_layers": [],
      "extensions": {},
      "features": {}
    }
  },
  "pci_info": {
    "device_pci_info": {
      "model": "Navi 23 [Radeon RX 6600/6600 XT/6600M]",
      "model_id": "73FF",
      "vendor": "Advanced Micro Devices, Inc. [AMD/ATI]",
      "vendor_id": "1002"
    },
    "subsystem_pci_info": {
      "model": "GIGABYTE RX 6600 Eagle",
      "model_id": "2334",
      "vendor": "Gigabyte Technology Co., Ltd",
      "vendor_id": "1458"
    }
  },
  "power_profile_modes": {
    "active": 1,
    "modes": {
      "0": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              5,
              1,
              0,
              4,
              800,
              4587520,
              -65536,
              0
            ]
          },
          {
            "clock_type": "SOCCLK",
            "values": [
              0,
              5,
              1,
              0,
              1,
              0,
              3276800,
              -6553,
              -6553
            ]
          },
          {
            "clock_type": "MEMCLK",
            "values": [
              0,
              5,
              1,
              0,
              4,
              800,
              327680,
              -65536,
              0
            ]
          }
        ],
        "name": "BOOTUP_DEFAULT"
      },
      "1": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              5,
              1,
              0,
              4,
              650,
              4587520,
              -3276,
              -65536
            ]
          },
          {
            "clock_type": "SOCCLK",
            "values": [
              0,
              5,
              1,
              0,
              1,
              0,
              655360,
              -6553,
              -6553
            ]
          },
          {
            "clock_type": "MEMCLK",
            "values": [
              0,
              5,
              4,
              850,
              4,
              800,
              327680,
              -65536,
              0
            ]
          }
        ],
        "name": "3D_FULL_SCREEN"
      },
      "2": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              5,
              1,
              0,
              3,
              0,
              5898240,
              -65536,
              0
            ]
          },
          {
            "clock_type": "SOCCLK",
            "values": [
              0,
              5,
              1,
              0,
              1,
              0,
              3407872,
              -6553,
              -6553
            ]
          },
          {
            "clock_type": "MEMCLK",
            "values": [
              0,
              5,
              1,
              0,
              3,
              0,
              1966080,
              -65536,
              0
            ]
          }
        ],
        "name": "POWER_SAVING"
      },
      "3": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              5,
              1,
              0,
              4,
              500,
              4587520,
              -65536,
              0
            ]
          },
          {
            "clock_type": "SOCCLK",
            "values": [
              0,
              5,
              1,
              0,
              1,
              0,
              3473408,
              -6553,
              -6553
            ]
          },
          {
            "clock_type": "MEMCLK",
            "values": [
              0,
              5,
              1,
              0,
              4,
              500,
              1966080,
              -65536,
              0
            ]
          }
        ],
        "name": "VIDEO"
      },
      "4": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              5,
              4,
              1000,
              1,
              0,
              3932160,
              0,
              0
            ]
          },
          {
            "clock_type": "SOCCLK",
            "values": [
              0,
              5,
              1,
              0,
              1,
              0,
              655360,
              -6553,
              -6553
            ]
          },
          {
            "clock_type": "MEMCLK",
            "values": [
              0,
              5,
              1,
              0,
              4,
              800,
              327680,
              -65536,
              0
            ]
          }
        ],
        "name": "VR"
      },
      "5": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              5,
              4,
              1000,
              1,
              0,
              3932160,
              0,
              0
            ]
          },
          {
            "clock_type": "SOCCLK",
            "values": [
              0,
              5,
              1,
              0,
              1,
              0,
              655360,
              -6553,
              -6553
            ]
          },
          {
            "clock_type": "MEMCLK",
            "values": [
              0,
              5,
              4,
              850,
              3,
              0,
              327680,
              -65536,
              -32768
            ]
          }
        ],
        "name": "COMPUTE"
      },
      "6": {
        "components": [
          {
            "clock_type": "GFXCLK",
            "values": [
              0,
              5,
              1,
              0,
              4,
              800,
              4587520,
              -65536,
              0
            ]
          },
          {
            "clock_type": "SOCCLK",
            "values": [
              0,
              5,
              1,
              0,
              1,
              0,
              3276800,
              -6553,
              -6553
            ]
          },
          {
            "clock_type": "MEMCLK",
            "values": [
              0,
              5,
              1,
              0,
              4,
              800,
              327680,
              -65536,
              0
            ]
          }
        ],
        "name": "CUSTOM"
      }
    },
    "value_names": [
      "FPS",
      "MinFreqType",
      "MinActiveFreqType",
      "MinActiveFreq",
      "BoosterFreqType",
      "BoosterFreq",
      "PD_Data_limit_c",
      "PD_Data_error_coeff",
      "PD_Data_error_rate_coeff"
    ]
  },
  "power_states": {
    "core": [
      {
        "enabled": true,
        "index": 0,
        "value": 500
      },
      {
        "enabled": true,
        "index": 1,
        "value": 2605
      },
      {
        "enabled": true,
        "index": 2,
        "value": 2700
      }
    ],
    "vram": [
      {
        "enabled": true,
        "index": 0,
        "value": 96
      },
      {
        "enabled": true,
        "index": 1,
        "value": 541
      },
      {
        "enabled": true,
        "index": 2,
        "value": 675
      },
      {
        "enabled": true,
        "index": 3,
        "value": 925
      }
    ]
  },
  "stats": {
    "busy_percent": 0,
    "clockspeed": {
      "gpu_clockspeed": 20,
      "vram_clockspeed": 96
    },
    "core_power_state": 1,
    "fan": {
      "control_enabled": false,
      "pmfw_info": {},
      "pwm_current": 0,
      "pwm_max": 255,
      "pwm_min": 0,
      "speed_current": 1395,
      "speed_max": 3100,
      "speed_min": 0
    },
    "memory_power_state": 0,
    "pcie_power_state": 1,
    "power": {
      "average": 3.0,
      "cap_current": 120.0,
      "cap_default": 100.0,
      "cap_max": 120.0,
      "cap_min": 94.0
    },
    "temps": {
      "edge": {
        "crit": 100.0,
        "crit_hyst": -273.1499938964844,
        "current": 28.0
      },
      "junction": {
        "crit": 110.0,
        "crit_hyst": -273.1499938964844,
        "current": 28.0
      },
      "mem": {
        "crit": 100.0,
        "crit_hyst": -273.1499938964844,
        "current": 26.0
      }
    },
    "voltage": {
      "gpu": 1000
    },
    "vram": {}
  }
}
